src/lib/state/state.svelte.ts

```diff
@@ -64,11 +67,66 @@ export const globalState = $state<GlobalState>({
     // Indicates if user is using existing approval process, creating one or injected provider
     approvalType: 'existing',
 
+    constructorArguments: {
+      values: {},
+      required: 0
+    },
+
+    deterministic: {
+      isSelected: false,
+      salt: undefined
+    },
+
     // Indicates if deployment is completed.
     completed: false,
   },
 });
 
+export const isValidFormAuthentication = () => globalState.authenticated
+
+const isFormFilledFor = (formInputToCheck: keyof typeof globalState.form) => () => Boolean(globalState.form[formInputToCheck])
+export const isValidFormNetwork = isFormFilledFor("network")
+
+export const isValidConstructorArguments = () => Object.keys(globalState.form.constructorArguments.values).length === globalState.form.constructorArguments.required
```

Comment:
If a user fills a constructor argument input and then removes the content, the arg is empty but this condition still true because the entry in the record exists. We should validate that all the inputs are filled (not empty or undefined values)